src/classes/challenge.py

```diff
@@ -1,17 +1,41 @@
 # pylint: disable=too-few-public-methods
-# pylint: disable=too-many-arguments
 class Challenge():
-    """Class for challenge object"""
-    def __init__(self,
-                 idx: int,
-                 title: str,
-                 category:str,
-                 description: str,
-                 pts: int,
-                 difficulty: str):
-        self.idx = idx
-        self.title = title
-        self.category = category
-        self.description = description
-        self.pts = pts
-        self.difficulty = difficulty
+    """Class for the Challenge object"""
+    def __init__(self, challenge_id: int, data: dict):
+        """
+            idx: int,
+            author_id: int,
+            title: str,
+            category:str,
+            description: str,
+            pts: int,
+            difficulty: str
+        """
+        self.idx = challenge_id
+
+        parsed_data = Challenge.parse_rootme_challenge_data(data)
+        (self.author_id, self.title, self.category, self.description, self.pts, self.difficulty) = parsed_data
+
+    @staticmethod
+    def parse_rootme_challenge_data(data):
+        """A static method that returns a tuple of few challenge data extracted from raw RootMe API data"""
+
+        # multitline extraction so that in case of error (API change for exemple) the error clearly show the wrong line
+        data = data[0]
```

Comment:
Check len and fail if len = 0?

Reply:
You can check the [RootMe API](https://api.www.root-me.org/challenges/1) it never returns data that isn't a list of len > 0. Even on error (challenge id not found for example)
If there was an error in the `RootMeAPIManager` requesting the wrong endpoint (and then not returning expected data) I think that this should:
- be tested by unit tests
- be dealt with in the `RootMeAPIManager` class

We can't check every data at every step anyway :smile: 
I think we can assume at this step that data has the appropriate expected format, and any case where it hasn't should be dealt at lower level (in the API Manager) or is an error from the developer (passing the wrong data, for example user data from `RootMeAPIManager.get_user_by_id`)